pkgs/by-name/be/beekeeper-studio/package.nix

```diff
@@ -0,0 +1,52 @@
+{ appimageTools
+, fetchurl
+, lib
+, makeWrapper
+, stdenv
+}:
+
+let
+  pname = "beekeeper-studio";
+  version = "4.0.3";
+
+  plat = {
+    aarch64-linux = "-arm64";
+    x86_64-linux = "";
+  }.${stdenv.hostPlatform.system};
+
+  hash = {
+    aarch64-linux = "sha256-PdRGZedNMZtGPI6Za2T94yKXvk6JoMoAot/kFKsBqQs=";
+    x86_64-linux = "sha256-P2HlCb0dai4cJlKcT9LLQ+njViD+oHS3KI2YdR5jR44=";
+  }.${stdenv.hostPlatform.system};
+
+  src = fetchurl {
+    url = "https://github.com/beekeeper-studio/beekeeper-studio/releases/download/v${version}/Beekeeper-Studio-${version}${plat}.AppImage";
+    inherit hash;
+  };
+
+  appimageContents = appimageTools.extractType2 { inherit pname version src; };
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+
+  extraInstallCommands = ''
+    mv $out/bin/{${pname}-${version},${pname}}
+    source "${makeWrapper}/nix-support/setup-hook"
+    wrapProgram $out/bin/${pname} \
+      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
+    install -Dm444 ${appimageContents}/${pname}.desktop -t $out/share/applications/
+    install -Dm444 ${appimageContents}/${pname}.png -t $out/share/pixmaps/
```

Comment:
Why change from `hicolor/512x512` to `pixmaps`? Is there a guide or standard i may view?

Reply:
idk, no reason rly, I just noticed that many packages with a single icon, copied from a bit random location usually have it installed in pixmaps in the end.

Also, personally - we don't know if the png in this location won't change its size in the future, making eg 256x256 location invalid. Unless if we can just put the icon in hicolor root 🤔?

Not sure what I should do here, perhaps someone could explain what's the best practice in this case?

I can revert this if you want

Reply:
Looks like icons/hicolor doesn't have any icons in root.

Also noticed that `file` can see resolutions, perhaps I could parse its output and then create appropriate directory.

Would be also a nice idea to have it solved for everyone by having some nativebuildinput/script/whatever you call it to just pass to it directory with icons/direct path to icon and it would create appropriate structure in $out.

unless if that's already solved and I'm blind.